src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,116 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double INCREASE_CAPACITY = 1.5;
+    private static final int ARRAYS_ZERO_BORDER = 0;
+    private static final int INCREASE_INDEX_FOR_COPY = 1;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (elementData.length == size) {
+            grow();
+        }
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (elementData.length == size) {
+            grow();
+        }
+        size++;
+        if (checkIndex(index)) {
+            System.arraycopy(elementData, index, elementData,
+                    index + INCREASE_INDEX_FOR_COPY,
+                    size - (index + INCREASE_INDEX_FOR_COPY));
+            elementData[index] = value;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Index out of bounds array size");
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (elementData.length < size + list.size()) {
+            grow();
+        }
```

Comment:
you have already do this check in the `add(T value)` method, do we need it here?